src/check_python_versions/sources/tox.py

```diff
@@ -99,13 +100,19 @@ def tox_env_to_py_version(env: str) -> Optional[Version]:
     If the environment name has dashes, only the first part is considered,
     e.g. py34-django20 becomes '3.4', and jython-docs becomes 'jython'.
     """
+    global has_dotted_env
+    has_dotted_env = False  # reset global state before check
     if '-' in env:
         # e.g. py34-coverage, pypy-subunit
         env = env.partition('-')[0]
     if env.startswith('pypy'):
         return Version.from_string('PyPy' + env[4:])
-    elif env.startswith('py') and len(env) >= 4 and env[2:].isdigit():
-        return Version.from_string(f'{env[2]}.{env[3:]}')
+    elif env.startswith('py'):
+        if len(env) >= 4 and env[2:].isdigit():
+            return Version.from_string(f'{env[2]}.{env[3:]}')
+        if '.' in env:
+            has_dotted_env = True
+            return Version.from_string(f'{env[2]}.{env[4:]}')
```

Comment:
We may have reached the point where manual substring checks are no longer simpler than a regular expression.

Besides, I think this mishandles corner cases like `py.29` (should return None) or `py10.20` (should return `Version('10.20')`).